cmd/waku/relay.go

```diff
@@ -0,0 +1,137 @@
+	for nodeTopic, cTopics := range pubSubTopicMap {
+		nodeTopic := nodeTopic
+		_, err := wakuNode.Relay().Subscribe(ctx, wprotocol.NewContentFilter(nodeTopic, cTopics...), relay.WithoutConsumer())
+		if err != nil {
+			return err
+		}
+
+		if len(options.Rendezvous.Nodes) != 0 {
+			// Register the node in rendezvous point
+			iter := rendezvous.NewRendezvousPointIterator(options.Rendezvous.Nodes)
+
+			wg.Add(1)
+			go func(nodeTopic string) {
+				t := time.NewTicker(rendezvous.RegisterDefaultTTL)
+				defer t.Stop()
+				defer wg.Done()
+
+				for {
+					select {
+					case <-ctx.Done():
+						return
+					case <-t.C:
+						// Register in rendezvous points periodically
+						wakuNode.Rendezvous().RegisterWithNamespace(ctx, nodeTopic, iter.RendezvousPoints())
+					}
+				}
+			}(nodeTopic)
+
+			wg.Add(1)
+			go func(nodeTopic string) {
+				defer wg.Done()
+				desiredOutDegree := wakuNode.Relay().Params().D
+				t := time.NewTicker(7 * time.Second)
+				defer t.Stop()
+				for {
+					select {
+					case <-ctx.Done():
+						return
+					case <-t.C:
+						peerCnt := len(wakuNode.Relay().PubSub().ListPeers(nodeTopic))
+						peersToFind := desiredOutDegree - peerCnt
+						if peersToFind <= 0 {
+							continue
+						}
+
+						rp := <-iter.Next(ctx)
+						if rp == nil {
+							continue
+						}
+						ctx, cancel := context.WithTimeout(ctx, 7*time.Second)
+						wakuNode.Rendezvous().DiscoverWithNamespace(ctx, nodeTopic, rp, peersToFind)
+						cancel()
+					}
+				}
+			}(nodeTopic)
+
+		}
+	}
+
+	// Bridge topics
+	bridgedTopics := make(map[string]map[string]struct{})
+	bridgedTopicsSet := make(map[string]struct{})
+	for _, topics := range options.Relay.BridgeTopics {
+		_, ok := bridgedTopics[topics.FromTopic]
+		if !ok {
+			bridgedTopics[topics.FromTopic] = make(map[string]struct{})
+		}
+
+		bridgedTopics[topics.FromTopic][topics.ToTopic] = struct{}{}
+		bridgedTopicsSet[topics.FromTopic] = struct{}{}
+		bridgedTopicsSet[topics.ToTopic] = struct{}{}
+	}
+
+	// Make sure all topics are subscribed
+	for _, topic := range maps.Keys(bridgedTopicsSet) {
+		if !wakuNode.Relay().IsSubscribed(topic) {
+			_, err := wakuNode.Relay().Subscribe(ctx, wprotocol.NewContentFilter(topic), relay.WithoutConsumer())
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	for fromTopic, toTopics := range bridgedTopics {
+		subscriptions, err := wakuNode.Relay().Subscribe(ctx, wprotocol.NewContentFilter(fromTopic))
+		if err != nil {
+			return err
+		}
+
+		topics := maps.Keys(toTopics)
+		for _, subscription := range subscriptions {
+			wg.Add(1)
+			go func(subscription *relay.Subscription, topics []string) {
+				defer wg.Done()
+				for env := range subscription.Ch {
+					for _, topic := range topics {
+						// HACK: We append a magic number here, just so the pubsub message ID will change
+						env.Message().Meta = append(env.Message().Meta, 255)
```

Comment:
@jm-clius @alrevuelta 
Notice that here i'm appending a magic number just to force the pubsub message to have a different ID.
In https://github.com/waku-org/nwaku/blob/52759faa0341e5463afe6fb8cf51027216cc5d3b/waku/waku_relay/message_id.nim#L31-L33 We are only taking into account the payload of the pubsub message, but not the topic, meaning that you cant' have the same message payload publicated in different pubsub topics.  If I don't do this change, then I cannot fwd the message to a different pubsub topic.

I wonder if we should instead modify the message id function to also take into account the pubsub topic of the message? I'm okay with not modifying it tho, as this bridge functionality is something that will be used only during the transition period from default pubsub topic to static shard in status, and will probably remove this code after two status releases.

Reply:
This is a very good point. In fact, the message id should in fact be the deterministic message hash defined in https://rfc.vac.dev/spec/14/#deterministic-message-hashing. Previously, though, we changed message ID to match go-waku because the impact of a message ID mismatch in a network is quite large. Perhaps the time has come though to bite the bullet and synchronise client upgrades for TWN?

Reply:
Yeah, i remember! it was very surprising how using different hashing function had such a large impact :worried: 
The only concern I have is that at this point we do not have a Waku message available yet, and we'd need to decode it from the protobuffer stored in the `data` attribute.

The fields i could see populated are these, based on the config we set for gossipsub:
```
message Message {
	optional string from = 1;
	optional bytes data = 2;    <--------------
	optional bytes seqno = 3;
        required string topic = 4;  <--------------
	optional bytes signature = 5;
	optional bytes key = 6;
}
```